apps/wakucanary/certsgenerator.nim

```diff
@@ -0,0 +1,23 @@
+import osproc, os
+
+# Function to generate a self-signed certificate
+proc generateSelfSignedCertificate*(certPath: string, keyPath: string) : int =
+  
+  # Ensure the OpenSSL is installed
+  if findExe("openssl") == "":
+    echo "OpenSSL is not installed or not in the PATH."
+    return 1
+
+  # Command to generate private key and cert
+  let 
+    cmd = "openssl req -x509 -newkey rsa:4096 -keyout " & keyPath & " -out " & certPath &
```

Comment:
Just wondering if we leave the `CN=CommonNameOrHostname` like this, it may cause problems in case a node is validating the certificate. Can we pick this up from either system's domain name or public IP?

Reply:
Good point!Jjust changed it :)

Now we use the public IP and if for some reason it cannot be retrieved then "127.0.0.1" is used.
With domain name is more problematic as we want for it to be automatic and not require any input from the user, so public IP is used instead.

WDYT?